File: VisualBasicAssignment/screen/kangtze/addfood.vb
Imports System.Data.SqlClient
Imports System.Drawing.Image
Imports System.Drawing.Imaging
Imports System.IO

Public Class addfood

    Private Sub Page_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Try
                Dim cat As List(Of Catergory) = db.Catergories.ToList()
                Me.cmbCategory.DataSource = cat
                Me.cmbCategory.ValueMember = "id"
                Me.cmbCategory.DisplayMember = "name"
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using
    End Sub

    Private Sub btnUpload_Click(sender As Object, e As EventArgs) Handles btnUpload.Click

        Try
            ofdImage.Filter = "Image Files (*.jpg;*.jpeg;.*.gif;)|*.jpg;*.jpeg;.*.gif"
            If ofdImage.ShowDialog() = DialogResult.OK Then
                pcbProduct.Image = New Bitmap(ofdImage.FileName)
            End If
        Catch ex As Exception
            Console.WriteLine("Exception caught: {0}", ex)
        End Try
    End Sub

    Private Sub btnCreate_Click(sender As Object, e As EventArgs) Handles btnCreate.Click

        Dim ms As New MemoryStream
        pcbProduct.Image.Save(ms, pcbProduct.Image.RawFormat)

        Dim f As New Food()
        f.add_date = DateTime.Now()
        f.catergory_id = cmbCategory.SelectedValue
        f.descriptions = Me.txtDesc.Text
        f.duty_staff_id = DatabaseConnections.GetLoginUser().user_id
        f.title = Me.txtName.Text
        f.price = CType(Me.txtPrice.Text, Double)
        f.stock = 100
        f.image = ms.ToArray

        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Try
                db.Foods.Add(f)
                db.SaveChanges()
                MessageBox.Show(
                    "Updates sucessful" + db.Foods.Count.ToString(),
                    "SUCCESS",
                    MessageBoxButtons.OK,
                    MessageBoxIcon.Information)
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using

    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        txtDesc.Clear()
        txtName.Clear()
        txtPrice.Clear()
    End Sub
End Class

File: VisualBasicAssignment/screen/kangtze/addfood.cs
using System;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicAssignment
{
    public partial class addfood
    {
        public addfood()
        {
            InitializeComponent();
            _btnUpload.Name = "btnUpload";
            _btnCreate.Name = "btnCreate";
            _btnCancel.Name = "btnCancel";
        }

        private void Page_Load(object sender, EventArgs e)
        {
            using (var db = new FoodShopEntities1())
            {
                try
                {
                    var cat = db.Catergories.ToList();
                    cmbCategory.DataSource = cat;
                    cmbCategory.ValueMember = "id";
                    cmbCategory.DisplayMember = "name";
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void btnUpload_Click(object sender, EventArgs e)
        {
            try
            {
                ofdImage.Filter = "Image Files (*.jpg;*.jpeg;.*.gif;)|*.jpg;*.jpeg;.*.gif";
                if (ofdImage.ShowDialog() == DialogResult.OK)
                {
                    pcbProduct.Image = new Bitmap(ofdImage.FileName);
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine("Exception caught: {0}", ex);
            }
        }

        private void btnCreate_Click(object sender, EventArgs e)
        {
            var ms = new MemoryStream();
            pcbProduct.Image.Save(ms, pcbProduct.Image.RawFormat);
            var f = new Food();
            f.add_date = DateTime.Now;
            f.catergory_id = (int?)cmbCategory.SelectedValue;
            f.descriptions = txtDesc.Text;
            f.duty_staff_id = DatabaseConnections.GetLoginUser().user_id;
            f.title = txtName.Text;
            f.price = Conversions.ToDouble(txtPrice.Text);
            f.stock = 100;
            f.image = ms.ToArray();
            using (var db = new FoodShopEntities1())
            {
                try
                {
                    db.Foods.Add(f);
                    db.SaveChanges();
                    MessageBox.Show("Updates sucessful" + db.Foods.Count().ToString(), "SUCCESS", MessageBoxButtons.OK, MessageBoxIcon.Information);
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            txtDesc.Clear();
            txtName.Clear();
            txtPrice.Clear();
        }
    }
}